### ISSUE TITLE
🐛 [TF] `create_optimizer` wrong superposition of learning rate schedules

### ISSUE THREAD

**Issue by Colanim** [huggingface/transformers#5098]
# 🐛 Bug

When using `create_optimizer`, 2 learning rate schedulers are placed on top of each other (WarmUp and keras Polynomial Decay) :

https://github.com/huggingface/transformers/blob/efeb75b8054cc299698cf8bc09f395ada2660745/src/transformers/optimization_tf.py#L70-L80

But the step in Warmup scheduler is not updated to take into account the warmup steps, which lead to a wrong learning rate :

![image](https://user-images.githubusercontent.com/43774355/84980216-69edd500-b16c-11ea-97f6-44642bdc64ce.png)

---

Expected learning rate shape :

![Dessin sans titre](https://user-images.githubusercontent.com/43774355/84981152-d2d64c80-b16e-11ea-9092-ccc50217fc66.png)
